app/services/unmute_service.rb

```diff
@@ -1,10 +1,14 @@
 # frozen_string_literal: true
 
 class UnmuteService < BaseService
-  def call(account, target_account)
-    return unless account.muting?(target_account)
+  def call(account, target_account, mute = nil)
+    if mute
+      mute.destroy!
+    else
+      return unless account.muting?(target_account)
 
-    account.unmute!(target_account)
+      account.unmute!(target_account)
+    end
```

Comment:
It makes sense, but since mutes are supposed to be unique per couple of accounts (this is enforced with a unique index in the database), I would just use `mute.account_id` and `mute.target_account_id` in the caller, this would avoid having to make changes to this service.